clients/src/main/java/org/apache/kafka/clients/admin/internals/AlterShareGroupOffsetsHandler.java

```diff
@@ -0,0 +1,182 @@
+                        .setPartitionIndex(tp.partition())
+                        .setStartOffset(startOffset)
+                );
+            }
+            requestTopics.add(new AlterShareGroupOffsetsRequestData.AlterShareGroupOffsetsRequestTopic()
+                    .setTopicName(topic)
+                    .setPartitions(partitionList));
+        }
+
+        AlterShareGroupOffsetsRequestData data = new AlterShareGroupOffsetsRequestData()
+                .setGroupId(groupId.idValue)
+                .setTopics(requestTopics);
+
+        return new AlterShareGroupOffsetsRequest.Builder(data);
+    }
+
+    public static AdminApiFuture.SimpleAdminApiFuture<CoordinatorKey, Map<TopicPartition, Errors>> newFuture(String groupId) {
+        return AdminApiFuture.forKeys(Collections.singleton(CoordinatorKey.byGroupId(groupId)));
+    }
+
+    @Override
+    public String apiName() {
+        return "alterShareGroupOffsets";
+    }
+
+    @Override
+    public ApiResult<CoordinatorKey, Map<TopicPartition, Errors>> handleResponse(Node broker, Set<CoordinatorKey> keys, AbstractResponse abstractResponse) {
+        AlterShareGroupOffsetsResponse response = (AlterShareGroupOffsetsResponse) abstractResponse;
+        final Map<TopicPartition, Errors> partitionResults = new HashMap<>();
+        final Set<CoordinatorKey> groupsToUnmap = new HashSet<>();
+        final Set<CoordinatorKey> groupsToRetry = new HashSet<>();
+
+        for (AlterShareGroupOffsetsResponseData.AlterShareGroupOffsetsResponseTopic topic : response.data().responses()) {
+            for (AlterShareGroupOffsetsResponseData.AlterShareGroupOffsetsResponsePartition partition : topic.partitions()) {
+                TopicPartition topicPartition = new TopicPartition(topic.topicName(), partition.partitionIndex());
+                Errors error = Errors.forCode(partition.errorCode());
+
+                if (error != Errors.NONE) {
+                    handleError(
+                            groupId,
+                            topicPartition,
+                            error,
+                            partitionResults,
+                            groupsToUnmap,
+                            groupsToRetry
+                    );
+                } else {
+                    partitionResults.put(topicPartition, error);
+                }
+            }
+        }
+
+        if (groupsToUnmap.isEmpty() && groupsToRetry.isEmpty()) {
+            return ApiResult.completed(groupId, partitionResults);
+        } else {
+            return ApiResult.unmapped(new ArrayList<>(groupsToUnmap));
+        }
+    }
+
+    private void handleError(
+            CoordinatorKey groupId,
+            TopicPartition topicPartition,
+            Errors error,
+            Map<TopicPartition, Errors> partitionResults,
+            Set<CoordinatorKey> groupsToUnmap,
+            Set<CoordinatorKey> groupsToRetry
+    ) {
+        switch (error) {
```

Comment:
KIP also defines `GROUP_AUTHORIZATION_FAILED` and `KAFKA_STORAGE_ERROR` do we need to handle that as well here?